test/GovernanceFactory.t.sol

```diff
@@ -119,8 +124,24 @@ contract GovernanceFactoryTest is Test {
         assertTrue(_governanceFactory.supportsInterface(ifId));
     }
 
-    function testSupportsInterfaceUpgradeable() public {
-        bytes4 ifId = type(Upgradeable).interfaceId;
+    function testSupportsInterfaceVersioned() public {
+        bytes4 ifId = type(Versioned).interfaceId;
         assertTrue(_governanceFactory.supportsInterface(ifId));
     }
+
+    function testProxyUpgrade() public {
```

Comment:
Can someone explain what this test is doing?

Reply:
1. views the proxy as an upgradeable
2. upgrades the proxy to a new version that implements isUpgraded
3. calls the function confirming the upgrade worked